test/unit/services/edition_publisher_test.rb

```diff
@@ -65,6 +65,19 @@ class EditionPublisherTest < ActiveSupport::TestCase
     assert_equal 1.week.ago, edition.major_change_published_at
   end
 
+  test '#perform! with an unpublished edition sets the government association' do
+    government = create(:government)
+    unpublished_edition = create(:unpublished_edition)
+    edition = unpublished_edition.create_draft(create(:policy_writer))
```

Comment:
Confusingly, an `unpublished_edition` is one that's been published before and since unpublished. I think `withdrawn_edition` would probably be better language for that, but that's a discussion for another place.

You want to use `create(:draft_edition)` here and then skip the `create_draft` call.